### DISCUSSION TITLE
Prism Uno Samples

### DISCUSSION THREAD

**Discussion by Alex-Dobrynin** [unoplatform/uno#5123] [Q&A] [answered]
Hello, I'm Xamarin.Forms developer and earlier was WPF dev. So, on WPF and Xamarin.Forms I worked with Prism as MVVM framework. I saw that Prism now supports Uno and I want to try this framework, but I'm new to UWP. I know how Prism's regions work, but I have doubts about Android's and UWP's hardware back button together with Prism's region navigation and how it is implemented within Uno.

I know that in UWP and Uno navigation made using Pages but in Prism it's made by UserControls and here I'm dissapointed.
It would be great that you give some samples in this direction so I can easily dive into the Uno world.

Also, I have question about MVVMCross and Uno and when it will be implemented?

**Comment by jeromelaban** [comment]
Hi Alex,

Currently, the Prism implementation for Uno is based on the WPF behavior, where there's no "back" navigation available as it is mostly meant for desktop applications.

We have not yet gone through implementating a navigation service that is similar to the one found for Xamarin.Forms, but I'm pretty sure the Prism team would be happy to have contributions for Uno and UWP/WinUI.

With regards to back navigation, apps can use [`SystemNavigationManager`](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Core.SystemNavigationManager?view=winrt-19041) class, with an [overview here](https://docs.microsoft.com/en-us/windows/uwp/design/basics/navigation-history-and-backwards-navigation).

**Reply by Alex-Dobrynin** [reply]
Thx for answer.
But what about some samples for Prism-Uno?

**Comment by davidjohnoliver** [comment, marked as answer]
The Uno Contoso sample app (an Uno port of the UWP Contoso app from Microsoft) uses Prism.

The code is here: https://github.com/unoplatform/Uno.Samples/tree/master/UI/UnoContoso

and there's a blog post with some more info here: https://platform.uno/blog/porting-the-contoso-uwp-showcase-application-to-uno-platform/

**Reply by Alex-Dobrynin** [reply]
Very grateful!
